### PR TITLE
feat: v2.3.0-rc0, clean and sort codebase, add documentation

### PR DESCRIPTION
**Pull request by enitrat**
- Adds documentation
- Cleans the codebase
- Sorts imports
- Adds memory function to set active context
- Bump to cairo v2.3.0
- Remove prelude dependencies
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/evm/src/helpers.cairo [hunk 1]
```diff
@@ -1,8 +1,14 @@
 use evm::errors::{EVMError, TYPE_CONVERSION_ERROR};
 
-// Try converting u256 to u32
-impl U256IntoResultU32 of Into<u256, Result<u32, EVMError>> {
-    fn into(self: u256) -> Result<u32, EVMError> {
+trait TryIntoResult<T, U> {
```
Comment (Eikix) [inline crates/evm/src/helpers.cairo:3]: gang gang

File: crates/evm/src/machine.cairo [hunk 2]
```diff
@@ -54,8 +59,19 @@ impl DefaultMachine of Default<Machine> {
...
+
 #[generate_trait]
-impl MachineCurrentContextImpl of MachineCurrentContext {
+impl MachineCurrentContextImpl of MachineCurrentContextTrait {
+    /// Sets the current execution context being executed by the machine.
+    /// This is an implementation-specific concept that is used
+    /// to divide a unique Stack/Memory simulated by a dict into
+    /// multiple sub-structures relative to a single context.
+    #[inline(always)]
+    fn set_active_execution_ctx(ref self: Machine, id: usize) {
```
Comment (Eikix) [inline crates/evm/src/machine.cairo:70]: then don't we need to change self.current_context?

Comment (Eikix) [reply crates/evm/src/machine.cairo:70]: `active_execution_context` as a param, and then,
memory/stack.set_active_segment(active_execution_context.context_id)

File: crates/evm/src/errors.cairo [hunk 3]
```diff
@@ -34,6 +35,8 @@ impl EVMErrorIntoU256 of Into<EVMError, u256> {
             EVMError::ReturnDataError(error_message) => error_message.into(),
             EVMError::JumpError(error_message) => error_message.into(),
             EVMError::NotImplemented => 'NotImplemented'.into(),
+            // TODO: refactor with dynamic strings once supported
+            EVMError::UnknownOpcode => 'UnknownOpcode'.into()
```
[the remark this answers is not in the document]
Comment (enitrat) [reply crates/evm/src/errors.cairo:39]: Unknown is different than not implemented

File: crates/evm/src/helpers.cairo (hunk 1 above)
Comment (enitrat) [reply crates/evm/src/helpers.cairo:3]: hmmm special trait so good

File: crates/evm/src/machine.cairo (hunk 2 above)
Comment (enitrat) [reply crates/evm/src/machine.cairo]: okay khrap

**Review by Eikix — DISMISSED** [review]
lgtm 🔥🔥🔥

File: crates/evm/src/errors.cairo (hunk 3 above)
[the remark this answers is not in the document]
Comment (Eikix) [reply crates/evm/src/errors.cairo:39]: I meant, print format!("Unknown Opcode {opcode}")

**Comment by enitrat** [comment]
@Eikix all good now

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**